components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -41,6 +50,22 @@ class DownloadsFeature(
      */
     fun start() {
         super.observeSelected()
+        fun reAttachOnStartDownloadListener(previousDialog: DownloadDialogFragment?) {
+            previousDialog?.apply {
+                this@DownloadsFeature.dialog = this
+                val selectedSession = sessionManager.selectedSession
+                val download = selectedSession?.download
+                download?.consume {
+                    onDownload(selectedSession, it)
+                    false
+                }
+            }
+        }
```

Comment:
I wonder why this is an inline function inside start()? At least in the diff it makes it hard to read. :)

Reply:
Much nicer!!
I pulled the inline function as a private class member
![image](https://user-images.githubusercontent.com/773158/47316780-e89c1300-d615-11e8-9ee8-a3fb987a2b99.png)